src/main.rs

```diff
@@ -30,45 +36,24 @@ async fn main() -> anyhow::Result<()> {
     let opts: Opts = Opts::parse();
 
     let pool = sqlx::PgPool::connect(&env::var("DATABASE_URL")?).await?;
+    
+    init_tracing();
 
-    let config = near_lake_framework::LakeConfigBuilder::default()
-        .s3_bucket_name(opts.s3_bucket_name)
-        .s3_region_name(opts.s3_region_name)
-        .start_block_height(opts.start_block_height)
-        .blocks_preload_pool_size(100)
-        .build()?;
-
-    let _writer_guard = init_tracing();
-
-    let (lake_handle, stream) = near_lake_framework::streamer(config);
-
-    let mut handlers = tokio_stream::wrappers::ReceiverStream::new(stream)
-        .map(|streamer_message| handle_streamer_message(streamer_message, &pool))
-        .buffer_unordered(1usize);
-
-    // let mut time_now = std::time::Instant::now();
-    while let Some(handle_message) = handlers.next().await {
-        match handle_message {
-            Ok(_block_height) => {
-                // let elapsed = time_now.elapsed();
-                // println!(
-                //     "Elapsed time spent on block {}: {:.3?}",
-                //     block_height, elapsed
-                // );
-                // time_now = std::time::Instant::now();
-            }
-            Err(e) => {
-                return Err(anyhow::anyhow!(e));
-            }
-        }
-    }
+    // create a lake configuration with S3 information passed in as ENV vars
+    let config = opts.get_lake_config().await;
```

Comment:
We call this method which returns the lake streamer configuration with the aws credentials passed in as env vars